runtime/jcl/common/system.c

```diff
@@ -182,9 +182,9 @@ Java_java_lang_System_getSysPropBeforePropertiesInitialized(JNIEnv *env, jclass
 		if (NULL == sysPropValue) {
 			/* As part of better handling of JEP400 constraints on z/OS, the com.ibm.autocvt property
 			 * determines whether we convert input file I/O based on file tagging. If not explicitly specified,
-			 * the property defaults to true, unless file.encoding is set to COMPAT. */
+			 * the property defaults to true, unless file.encoding is set. */
 			const char *fileEncodingValue = getDefinedArgumentFromJavaVMInitArgs(vmInitArgs, "file.encoding");
-			if ((NULL != fileEncodingValue) && (0 == strcmp("COMPAT", fileEncodingValue))) {
+			if (NULL != fileEncodingValue) {
 				sysPropValue = "false";
 			} else {
 				sysPropValue = "true";
```

Comment:
This isn't (or shouldn't be) just about "input", encoding is also an "output" consideration. The `if` test shrunk, so a ternary operator seems appropriate.
```c
			/* As part of better handling of JEP400 constraints on z/OS, the com.ibm.autocvt property
			 * determines whether file I/O considers file tagging. If not explicitly specified,
			 * the property defaults to true, unless file.encoding is set. */
			const char *fileEncodingValue = getDefinedArgumentFromJavaVMInitArgs(vmInitArgs, "file.encoding");
			sysPropValue = (NULL == fileEncodingValue) ? "true" : "false";
```

Reply:
Good point.  Updated as suggested.  Thanks!